### PR TITLE
Add elliptic eigenvalue global prom example

### PR DESCRIPTION
**Pull request by ckendrick**
Adds a simple elliptic eigenvalue global pROM example based on [MFEM example 11p](https://github.com/mfem/mfem/blob/master/examples/ex11p.cpp).

**Note**: MFEM needs to be compiled with `MFEM_USE_LAPACK` enabled to run the online phase. This adds a `-l` option to the `compile.sh` script to build MFEM with lapack enabled.

Sample runs:
```
./elliptic_eigenproblem_global_rom -offline -p 2 -id 0 -a 0.40
./elliptic_eigenproblem_global_rom -offline -p 2 -id 1 -a 0.45
./elliptic_eigenproblem_global_rom -offline -p 2 -id 2 -a 0.55
./elliptic_eigenproblem_global_rom -offline -p 2 -id 3 -a 0.60

./elliptic_eigenproblem_global_rom -merge -p 2 -ns 4

./elliptic_eigenproblem_global_rom -fom -p 2 -a 0.50 -visit

./elliptic_eigenproblem_global_rom -online -p 2 -a 0.50
```

Example FOM output:
```
Number of unknowns: 289
 Eigenvalue 0: 0.04533314
 Eigenvalue 1: 0.11332411
 Eigenvalue 2: 0.11486387
 Eigenvalue 3: 0.18192
 Eigenvalue 4: 0.22964377
Elapsed time for assembling FOM: 1.475750e-03 second
Elapsed time for solving FOM: 3.390041e-01 second
```

Example online output:
```
spatial basis dimension is 289 x 20
Eigenvalue 0: = 0.048430949
Eigenvalue 1: = 0.12021157
Eigenvalue 2: = 0.12147847
Eigenvalue 3: = 0.19456504
Eigenvalue 4: = 0.24285855
Relative error of ROM solution for eigenvalue 0 = 0.068334316
Relative error of ROM solution for eigenvalue 1 = 0.060776586
Relative error of ROM solution for eigenvalue 2 = 0.057586388
Relative error of ROM solution for eigenvalue 3 = 0.069508795
Relative error of ROM solution for eigenvalue 4 = 0.057544708
Relative l2 error of ROM eigenvector 0 = 0.070430601
Relative l2 error of ROM eigenvector 1 = 1.996413
Relative l2 error of ROM eigenvector 2 = 0.077548522
Relative l2 error of ROM eigenvector 3 = 0.10685373
Relative l2 error of ROM eigenvector 4 = 0.1494834
Elapsed time for assembling ROM: 5.230334e-03 second
Elapsed time for solving ROM: 7.794170e-04 second
```

### CODE REVIEW

File: scripts/compile.sh
Comment (siuwuncheung) [inline scripts/compile.sh:29]: I'm curious if there are any reasons to have different kinds of options, `"On"` vs `"Off"`, `"Yes"` vs `"No"`, `true` vs `false`. Can we unify them?

**Comment by siuwuncheung** [comment]
Inviting @dreamer2368 as a reviewer to evaluate the changes in the build process in this PR.

**Review by dreamer2368 — COMMENTED** [review]
The setup procedure looks good in general. If LAPACK is required for this example, can we install mfem with LAPACK dependency in our docker container?

File: examples/prom/elliptic_eigenproblem_global_rom.cpp
Comment (siuwuncheung) [inline examples/prom/elliptic_eigenproblem_global_rom.cpp:35]: When you get a chance, would you please update the command line options, timing and error in the PR description and the header of the script? I think `-a 1.1` and `-a 1.2` in the offline phase do not work anymore, and it might be `-a` instead of `-f` in the online phase.

Comment (siuwuncheung) [reply examples/prom/elliptic_eigenproblem_global_rom.cpp:35]: Update: I saw that the new command lines and results in PR description, and the new results in the header. It remains to correct the command lines in the header.

File: examples/prom/elliptic_eigenproblem_global_rom.cpp
Comment (siuwuncheung) [inline examples/prom/elliptic_eigenproblem_global_rom.cpp:416]: We need to do `delete superlu`?

File: examples/prom/elliptic_eigenproblem_global_rom.cpp
Comment (siuwuncheung) [inline examples/prom/elliptic_eigenproblem_global_rom.cpp:407]: We need to do `delete amg`?

File: examples/prom/elliptic_eigenproblem_global_rom.cpp
Comment (siuwuncheung) [inline examples/prom/elliptic_eigenproblem_global_rom.cpp:427]: We need to do `delete strumpack`?

File: examples/prom/elliptic_eigenproblem_global_rom.cpp
Comment (siuwuncheung) [inline examples/prom/elliptic_eigenproblem_global_rom.cpp:442]: We need to do `delete cpardiso`?

**Review by siuwuncheung — COMMENTED** [review]
LGTM except some minor things. Thanks @ckendrick!

File: examples/prom/elliptic_eigenproblem_global_rom.cpp
Comment (ckendrick) [reply examples/prom/elliptic_eigenproblem_global_rom.cpp:407]: I think this should be taken care of already since `amg`,`superlu`, etc are assigned to `precond`, which gets deleted here
https://github.com/LLNL/libROM/blob/92c3c4531a9826460b05be5219749446d38957f8/examples/prom/elliptic_eigenproblem_global_rom.cpp#L491

**Review by siuwuncheung — APPROVED** [review]

**Comment by siuwuncheung** [comment]
I had approved this PR after reviewing the changes and reproducing the resutls. We need one more approval to get this PR merged.

File: lib/mfem/Utilities.cpp
Comment (dylan-copeland) [inline lib/mfem/Utilities.cpp:79]: It looks like the condition being checked here is
`if (std::abs(t(i) - mesh_center) < 0.5 * domain_limit)`
The current version is equivalent, but less clear.

**Review by dylan-copeland — APPROVED** [review]

**Pull request merged**
